src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -191,30 +201,67 @@ const std::shared_ptr<AbstractOperator>& SQLPipelineStatement::get_physical_plan
   return _physical_plan;
 }
 
-const std::vector<std::shared_ptr<OperatorTask>>& SQLPipelineStatement::get_tasks() {
+const std::vector<std::shared_ptr<AbstractTask>>& SQLPipelineStatement::get_tasks() {
   if (!_tasks.empty()) {
     return _tasks;
   }
 
-  _tasks = OperatorTask::make_tasks_from_operator(get_physical_plan(), _cleanup_temporaries);
+  if (_is_transaction_statement) {
+    _tasks = _get_transaction_tasks();
+  } else {
+    auto operator_tasks = OperatorTask::make_tasks_from_operator(get_physical_plan(), _cleanup_temporaries);
+    _tasks = std::vector<std::shared_ptr<AbstractTask>>(operator_tasks.cbegin(), operator_tasks.cend());
+  }
+
   return _tasks;
 }
 
+std::vector<std::shared_ptr<AbstractTask>> SQLPipelineStatement::_get_transaction_tasks() {
+  auto sql_statement = get_parsed_sql_statement();
+  const std::vector<hsql::SQLStatement*>& statements = sql_statement->getStatements();
+  auto* transaction_statement = reinterpret_cast<hsql::TransactionStatement*>(statements.front());
+  switch (transaction_statement->command) {
+    case hsql::kBeginTransaction:
+      return {std::make_shared<JobTask>([this] {
+        if (!_transaction_context || !_transaction_context->is_auto_commit())
+          this->set_warning_message(std::string("WARNING: Cannot begin transaction inside an active transaction."));
```

Comment:
This error message does not seem to match the check. Are these cases tested?

Reply:
The code is correct but complicated, because SQLPipelineStatement::get_physical_plan() is called before get_transaction_tasks() (to _precheck_ddl_operators) und requires a TransactionContext.
As a result, when get_transaction_tasks() is called, there may exist an automatically created auto-commit-transaction-context.

There are two possibilities to make the code easier:
1) use an extra switch in get_result_table to check for transaction statements before _precheck_ddl_operators is called
2) move _precheck_ddl_operators inside get_tasks(). I currently do not understand why it is outside.

Reply:
Done (2.).